src/families/cosmos/types.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+
+import type { BigNumber } from "bignumber.js";
+import type {
+  TransactionCommon,
+  TransactionCommonRaw
+} from "../../types/transaction";
+
+export type CoreStatics = {};
+export type CoreAccountSpecifics = {};
+export type CoreOperationSpecifics = {};
+export type CoreCurrencySpecifics = {};
+
+export type CosmosDelegation = {|
+  validatorAddress: string,
+  amount: BigNumber,
+  pendingRewards: BigNumber,
+  isActive: boolean, // TBD is there more status case? if so, switch to status field with an enum string.
+|};
+
+export type CosmosRedelegation = {|
```

Comment:
On the network level, redelegations are stored as `(delegator,sourceValidator,destValidator) -> [(amount, completionTime)....]`: the `completionTime,Amount` doesn't have to be unique for a `(delegator, sourceValidator, destValidator)`. It's okay if we unroll the list into a list of this type, we just have to be sure to catch it

Reply:
i'm not sure to really understand this part. @henri-ly can you help me reiterating on the type here? thanks!

Reply:
@henri-ly basically I'm saying : "If we're flattening the `entries` part of redelegation, no problem, but we must remember that this flattening must happen`

Reply:
we received this data from libcore in this kind of shape : 
`{
delegator: string,
sourceValidator: string,
destValidator: string,
entries: [{
 amount: number,
 completionTime: date
}]
}`

I choose to flatten it to simplify the UI part, so we will have to flatten when we received it from libcore

Reply:
ok thanks! let's keep this discussion in mind for future iteration. i'll take a safe path for now which is to not introduce yet the redelegations array until we are sure about the data model and also how design planned to address it

we'll come back to it in next days/weeks